### PR TITLE
Some improvements in tests #3:

### PR DESCRIPTION
**Pull request by frankmayer**
### Summary of Changes
- call static methods correctly

### Testing Instructions
Code review only.

### Documentation Changes Required
None

### CODE REVIEW

**Comment by frankmayer** [comment]
@mbabker IMHO since we are testing static methods, we should call them the correct way. Also, this way someone reading the code knows that it is a static method being called. It just seems right.

**Comment by mbabker** [comment]
Then you've honestly got a lot to do because all of the `$this->assert*` calls are to static methods 😉

Honestly, I think this is just one of those preference things.  When you've already got an object, using a static accessor like that looks really goofy (and without tweaks in PHP syntax you end up with issues, see the Travis report on that).  In cases where you don't already have an object or don't need an object, I'm all for the static method calls being made correctly.

```php
$foo = new Foo;

// Case 1
$foo->bar();
$foo::car();
$foo->far();
$foo::$jar;

// Case 2
$foo->bar();
Foo::car(); // or $foo->car();
$foo->far();
Foo::$jar;
```

To me when reading Case 1 it's just too easy to get yourself confused on what's going on.  Case 2 to me is much more readable and seems less error prone.

**Comment by frankmayer** [comment]
:smile:  yes, I see your point, but those are the methods of PHPUnit - won't/can't do anything for those.
But I can do for the ones being tested.

**Comment by frankmayer** [comment]
@mbabker Sorry, didn't reply on your second half of your comment. 
Yes, you're right.
However, since in most of the tests, the class where this method is called from, is changed depending on the inheriting test class, the options are these:

1) call the static methods appropriately (by moving the affected test-methods from the parent to the subclasses and calling the tested methods like `Foo::bar()` correctly there, instead of using the varying object.)
2) leaving it, as it is, with the changes of the current commit of this PR
3) Scrap this PR

**Comment by mbabker** [comment]
For `TestCaseCache` it needs that dynamic element, no argument there (otherwise why bother with a generic test case most of the cache adapters can inherit?)

For the stuff in the application classes, I'd just change it to `JApplicationX::getRouter()` as we know what class is being tested there and what the expectation should be, I don't think there's any reason to call it as `$object::getRouter()` there.

For the `JAccess` tests, I'd just remove instantiating a new instance of the class in full and call those as `JAccess::xx()` as everything in that class is static so we're just wasting cycles instantiating an object just to call its method for testing (in a way that doesn't match production use at that).

For the most part, unless there is a critical need for a dynamic base object to call a specific static method, I think we're better off just calling it using the full class name.  There are honestly only a few cases where a subclass might have an extended version and subclasses don't need to test the parent methods if they aren't extending them in any way.

File: tests/unit/suites/libraries/cms/application/JApplicationSiteTest.php [hunk 1]
```diff
@@ -229,7 +229,9 @@ public function testGetPathway()
 	 */
 	public function testGetRouter()
 	{
-		$this->assertInstanceOf('JRouterSite', $this->class->getRouter());
+		$class = $this->class;
```
Comment (andrepereiradasilva) [inline tests/unit/suites/libraries/cms/application/JApplicationSiteTest.php]: why do we need this now?

File: tests/unit/suites/libraries/joomla/session/JSessionTest.php [hunk 2]
```diff
@@ -210,7 +210,10 @@ public function testGetFormToken()
 		$user = JFactory::getUser();
 
 		$expected = md5($user->get('id', 0) . $this->object->getToken(false));
-		$this->assertEquals($expected, $this->object->getFormToken(false), 'Form token should be calculated as above.');
+
+			$object = $this->object;
```
Comment (andrepereiradasilva) [inline tests/unit/suites/libraries/joomla/session/JSessionTest.php]: cs: please remove one tab here

**Comment by mbabker** [comment]
Looks fine after the two inline comments are addressed.

File: tests/unit/suites/libraries/cms/application/JApplicationSiteTest.php (hunk 1 above)
Comment (frankmayer) [inline tests/unit/suites/libraries/cms/application/JApplicationSiteTest.php]: Missed that. Thanks!

**Review by andrepereiradasilva — APPROVED** [review]

**Comment by wilsonge** [comment]
👍

**Pull request merged**